investos/portfolio/risk_model/factor_risk.py

```diff
@@ -1,23 +1,351 @@
+    "debt_total_to_market_cap_open_dil_prev_1q": "d862bc25-e085-442a-acfe-05c880da2d40",  # 'prev_1q debt_total to market_cap_open_dil multiple. ID is FactSet Sym ID.'
+    # Quality
+    "net_income_div_sales_ltm": "7f1e058f-46b6-406f-81a1-d8a5b81371a2",  # ltm profit margin (as % of sales)
+    "ebit_div_sales_ltm": "47d7d1b0-46cc-4620-b0d3-35aa33fe490a",
+    # Value
+    "ev_to_ebit_ltm": "f65f7b6a-5433-4543-8f15-5f50e43dd9f9",
+    "market_cap_open_dil_to_net_income_ltm": "7e4264a9-a137-4f5b-bc10-1f1e90d2f3f5",
+    "market_cap_open_dil_to_operating_excl_wc_cf_ltm": "5f050fce-5099-4ce9-a737-5d8f536c5826",  # ltm market_cap_open_dil to operating_excl_wc_cf multiple. ID is FactSet Sym ID.
+    # Growth
+    "sales_ltm_growth_over_sales_ltm_lag_1a": "e6035b0a-65b9-409d-a02e-b3d47ca422e2",
+}
 
 
 class FactorRisk(BaseRisk):
     """Multi-factor risk model."""
 
     def __init__(
-        self, factor_covariance, factor_loadings, idiosyncratic_variance, **kwargs
+        self,
+        start_date,
+        end_date,
+        risk_model_window: int,
+        recalc_freq: int,
+        fos_risk_factor_uuids: list = [],
+        fos_return_uuid: str = "return_1d",
+        adjustments: dict = {"FactorCovEstArgs": {"window": 21, "half_life": 360}},
+        factor_covariance=None,
+        factor_loadings=None,
+        idiosyncratic_variance=None,
+        **kwargs,
     ):
         super().__init__(**kwargs)
 
-        self.factor_covariance = factor_covariance
-        self.factor_loadings = factor_loadings
-        self.idiosyncratic_variance = idiosyncratic_variance
+        self._risk_model_window = risk_model_window
+        self._recalc_freq = recalc_freq
+
+        self._fos_risk_factor_uuids = fos_risk_factor_uuids
+        self._fos_return_uuid = fos_return_uuid
+
+        self._end_date = end_date
+        self._start_date = start_date
+        self._adjustments = adjustments
+
+        if self._fos_risk_factor_uuids:
+            print("Running Barra Risk model...")
+            self._pull_fos_risk_factor_dfs()
+            self._generate_risk_models()
+            self._apply_risk_model_adjustments()
+            print(
+                "\nDone generating factor loadings,factor covariance, and idiosyncratic variance"
+            )
+        else:
+            self._factor_covariance = factor_covariance
+            self._factor_loadings = factor_loadings
+            self._idiosyncratic_variance = idiosyncratic_variance
+
+    # Initialization
+    def _pull_fos_risk_factor_dfs(self):
+        # pull from feature hub
+        print("\n1. Pulling out data from FeatureHub...")
+
+        if not fos.api_key:
+            raise ValueError(
+                "Feature hub connection failed. Please check the API key is valid."
+            )
+
+        # Initialize a DataFrame
+        self._df_loadings = (
+            fos.Feature.get(fos_return_uuid_dict[self._fos_return_uuid])
```

Comment:
Use self._fos_return_uuid here.

Reply:
You should add (and at the end take out) returns to the loadings df last, since it won't stay in the final loadings df. 

I think it's confusing to start with something that isn't actually a loading

Reply:
Hmm actually it's a good base df to start from... ignore me. Let's start with it